src/js/components/Table/Paginated/PaginatedTable.tsx

```diff
@@ -451,30 +464,55 @@ const PaginatedTable: PaginatedTableProps = ({
           filterOpen ? '' : 'u-hidden'
         } u-padSm u-spaceTopSm u-borderNeutral4 u-bgNeutral1 Grid-cell u-flex`}
       >
-        <FilterContext.Provider value={{ activeFilters, setActiveFilters }}>
-          <div className="u-size7of8">
-            {!isResettingFilters &&
-              filters.map((Item, index) => (
+        {useExternalFilterProvider ? (
+          <>
+            <div className="u-size7of8">
+              {filters.map((Item, index) => (
                 <Item key={index} isLast={index === filters.length - 1} />
               ))}
-          </div>
-          <div className="u-size1of8 u-textRight u-spaceRightMd">
-            <Button
-              type="text"
-              onClick={() => {
-                setActiveFilters({});
-                // This is a bit weird.
-                // We were seeing issues updating the components successfully when clearing out all the filter
-                // values. This, paired with the condition in the filter rendering, forces the filter section
-                // to rerender when we clear out all the filters, ensuring that its rendering with the "freshest"
-                // of values from the context.
-                setIsResettingFilters(true);
-              }}
-            >
-              Clear All
-            </Button>
-          </div>
-        </FilterContext.Provider>
+            </div>
+            <div className="u-size1of8 u-textRight u-spaceRightMd">
+              <Button
+                type="text"
+                onClick={() => {
+                  setActiveFilters({});
+                  // This is a bit weird.
+                  // We were seeing issues updating the components successfully when clearing out all the filter
+                  // values. This, paired with the condition in the filter rendering, forces the filter section
+                  // to rerender when we clear out all the filters, ensuring that its rendering with the "freshest"
+                  // of values from the context.
+                  setIsResettingFilters(true);
+                }}
+              >
+                Clear All
+              </Button>
+            </div>
+          </>
+        ) : (
+          <FilterContext.Provider value={{ activeFilters, setActiveFilters }}>
+            <div className="u-size7of8">
+              {filters.map((Item, index) => (
+                <Item key={index} isLast={index === filters.length - 1} />
+              ))}
+            </div>
+            <div className="u-size1of8 u-textRight u-spaceRightMd">
+              <Button
```

Comment:
What do you think about wrapping this code into another func? If you do that, you won't need to duplicate those lines... It would be like this:
```
<FilterContext.Provider value={{ activeFilters, setActiveFilters }}>
    {getFiltersAndButtons()}
</FilterContext.Provider>
```